### ISSUE TITLE
UnboundLocalError: local variable 'vrf_dict' referenced before assignment

### ISSUE THREAD

**Issue by sogloarcadius** [CiscoTestAutomation/pyats#38]
Hello,

I am facing an issue when using pyats and genie to parse VRF config on cisco CSR1000v device (iosxe).

```
pyats : 19.12
genie : 19.12

```

In fact the issue occurs when the VRF is being deleted by the router.

Below the output of the device of the command `show vrf detail`

```
VRF DEMO (VRF Id = 12); default RD 65001:1; default VPNID <not set>; being deleted
  Description: demo
  New CLI format, supports multiple address-families
  Flags: 0x180D
  No interfaces
Address family ipv4 unicast (Table ID = 0xC); being deleted:
  Flags: 0x1
  No Export VPN route-target communities
  No Import VPN route-target communities
  No import route-map
  No global export route-map
  No export route-map
  VRF label distribution protocol: not configured
  VRF label allocation mode: per-prefix
Address family ipv6 unicast not active
Address family ipv4 multicast not active
Address family ipv6 multicast not active


* Being deleted

```
After analysing the exception raised by the genie parser, i identify the issue is due to the fact that the regex used to parse the VRF details output fails for the first attributes parsing but succeed for the desription line causing an UnboundLocalError on the variable vrf_dict because the variable was not initialzed at the top level of the file.

FInd below the execption raised 

```

            # Description: desc
            m = p12.match(line)
            if m:
                groups = m.groupdict()
>               vrf_dict.update({'description': groups['desc']})
E               UnboundLocalError: local variable 'vrf_dict' referenced before assignment

../../../.virtualenvs/.../lib/python3.6/site-packages/genie/libs/parser/iosxe/show_vrf.py:464: UnboundLocalError

```
Is it possible to review the parser to consider cases when Cisco output VRF being deleted  like in my scenario.

In fact, the VRF parser should not try to retrieve the description if it fails to get the VRF name or the vrf_dict variable should be initialized at the top level of the file.

Any suggestion ?

Thanks,

Best regards,

**Comment by thaixuann** [comment]
Hello @sogloarcadius ,
Thanks for reporting the issue. We will take a look and get back to you.

--Thai.

**Comment by xiaoxinz-cisco** [comment]
Hello @sogloarcadius 

We fixed the issue and genie package **v20.1.1b0** has been released with the fix.
Could you please run the following command and try again to see if it works?

**pip install genie.libs.parser --upgrade --pre**

Please try and let's know if there is an issue.

Thanks,
Irene

**Comment by sogloarcadius** [comment]
Hello,

Using again pyats + genie  and I can confirm that my issue is fixed in the latest release.
Also the parser code look great now :)

Best regards,
Thanks

Arcadius.
